tests/buffer_test.py

```diff
@@ -0,0 +1,101 @@
+import imp
+import os
+import sys
+from click.testing import CliRunner
+from unittest import TestCase
+from swsssdk import ConfigDBConnector
+
+from .mock_tables import dbconnector
+
+import show.main as show
+import config.main as config
+from utilities_common.db import Db
+
+from .buffer_input.buffer_test_vectors import *
+
+test_path = os.path.dirname(os.path.abspath(__file__))
+modules_path = os.path.dirname(test_path)
+scripts_path = os.path.join(modules_path, "scripts")
+sys.path.insert(0, test_path)
+sys.path.insert(0, modules_path)
+
+class TestBuffer(object):
+    @classmethod
+    def setup_class(cls):
+        os.environ["PATH"] += os.pathsep + scripts_path
+        os.environ['UTILITIES_UNIT_TESTING'] = "2"
+        print("SETUP")
+
+    def setUp(self):
+        self.runner = CliRunner()
+        self.config_db = ConfigDBConnector()
+        self.config_db.connect()
+        self.obj = {'db': self.config_db}
+
+    def test_config_buffer_profile_headroom(self):
+        runner = CliRunner()
+        result = runner.invoke(config.config.commands["buffer"].commands["profile"].commands["add"],
+                               ["testprofile", "--dynamic_th", "3", "--xon", "18432", "--xoff", "32768"])
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+        self.executor(testData['show_buffer_config_headroom'])
```

Comment:
don't see 'testprofile' in the testData dict

Reply:
this is because the show test doesn't run on the updated db but the original mocked table.
tend to verify the content of the db in the configuration test cases and add a separate testcase for the show command

Reply:
Done.